website/server/controllers/api-v3/tasks/groups.js

```diff
@@ -16,6 +16,14 @@ import {
 
 let requiredGroupFields = '_id leader tasksOrder name';
 let types = Tasks.tasksTypes.map(type => `${type}s`);
+
+function canNotEditTasks(group, user, assignedUserId) {
+  let isNotGroupLeader = group.leader !== user._id;
+  let isManager = Boolean(group.managers[user._id]);
+  let userIsAssigningToSelf = Boolean(assignedUserId && user._id === assignedUserId);
+  let canNotEditTasks = isNotGroupLeader && !isManager && !userIsAssigningToSelf;
```

Comment:
Ah, sorry, that makes sense. I was confused by the mix of editing and assigning terminology here.

Is this line a concern here?
`let userIsAssigningToSelf = Boolean(assignedUserId && user._id === assignedUserId);`
Since the canNotEditTasks() function is used to control editing as well as assigning, will that line allow an ordinary member to edit the task if they are assigning it to themself? Even if the code is safe now, is it possible it could introduce that bug in future if the function is used in ways that it's not being used now? It feels odd that a function labelled as being for controlling editing is also being used to allow other permissions to users who should never be allowed to edit.

Would it be safer to have two functions, one to control editing and the other to control assigning? The function to control assigning could use canNotEditTasks() for most of its checks. At the least, I think this function needs to be renamed or commented to accurately describe what it's for.